Src/IronPython/Compiler/Parser.cs

```diff
@@ -635,13 +636,22 @@ public class Parser : IDisposable { // TODO: remove IDisposable
                     ReportSyntaxError(right.StartIndex, right.EndIndex, "starred assignment target must be in a list or tuple");
                 }
 
-                left.Add(right);
+                if (singleLeft == null) {
+                    singleLeft = right;
+                } else {
+                    if (left == null) {
+                        left = new List<Expression> { singleLeft };
+                    }
+
+                    left.Add(right);
+                }
+
                 right = MaybeEat(TokenKind.KeywordYield) ? ParseYieldExpression() : ParseTestListStarExpr();
             }
 
             CheckNotAssignmentTargetOnly(right);
 
-            var target = left.ToArray();
+            var target = left?.ToArray() ?? new [] { singleLeft };
```

Comment:
Looking at the method without context it wasn't clear that `singleLeft` is never null at this point. I wonder if it's worth adding `Debug.Assert(singleLeft != null)`. Maybe also `Debug.Assert(PeekToken(TokenKind.Assign))` at the top of the method?